src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java

```diff
@@ -234,9 +234,9 @@ private void mergeUpdatedFieldsToCaseDetails(final Map<String, JsonNode> data,
     }
 
     private void updateCaseState(CaseDetails caseDetails, CaseEventDefinition caseEventDefinition) {
-        Optional<String> postState = caseStateUpdateService.retrieveCaseState(caseEventDefinition, caseDetails);
-        if (postState.isPresent() && !equalsIgnoreCase(CaseStateDefinition.ANY, postState.get())) {
-            caseDetails.setState(postState.get());
+        String postState = caseStateUpdateService.retrieveCaseState(caseEventDefinition, caseDetails);
+        if (!equalsIgnoreCase(CaseStateDefinition.ANY, postState)) {
```

Comment:
if(!shouldRemainOnCurrentState(postState)){

where `shouldRemainOnCurrentState(String state)` is a private method containing `equalsIgnoreCase(CaseStateDefinition.ANY, postState)`

Reply:
suggesting also the same refactor on CreateCaseEventService